components/layout/construct.rs

```diff
@@ -198,8 +198,17 @@ impl InlineFragmentsAccumulator {
 
         match enclosing_style {
             Some(enclosing_style) => {
+                let mut frag_idx = 0;
+                let frag_len = fragments.len();
                 for frag in fragments.iter_mut() {
```

Comment:
I'd change this to `for (frag_idx, frag) in fragments.iter_mut().enumerate() {` and eliminate the `frag_idx` temporary.